core/src/main/scala/akka/kafka/internal/ConsumerStage.scala

```diff
@@ -190,7 +192,8 @@ private[kafka] object ConsumerStage {
       if (offsets.isEmpty)
         Future.successful(Done)
       else {
-        stages.head._2.commit(this)
+        implicit val ec = akka.dispatch.ExecutionContexts.global()
+        Future.sequence(stages.map(_._2.commit(this))).map(_ => Done)
```

Comment:
As far as I understand, the `sequence` call does a serie of `flatMap`s on futures and concatenates results into a sequence, which is a safe operation to be handled by the global EC. Can someone verify this? :)